From 6517c88f12d1d87225fa8a195c74d3e1f23da9f9 Mon Sep 17 00:00:00 2001
From: haiqiang yao <haiqiangx.yao@intel.com>
Date: Wed, 16 Mar 2016 18:35:22 +0800
Subject: [PATCH 2/5] atomisp.h:Add new parameter tdf for isp2.7

this patch will add new parameter tdf for isp2.7
Depends on fw patch: https://icggerrit.ir.intel.com/#/c/107991/1

Change-Id: I99234497047b68196a24c10f39dc9532160f7a4c
Orig-Tracked-On: https://jira01.devtools.intel.com/browse/OAM-2344
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-25608
Signed-off-by: haiqiang yao <haiqiangx.yao@intel.com>
Reviewed-on: https://android.intel.com:443/483549
---
 include/linux/atomisp.h | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/include/linux/atomisp.h b/include/linux/atomisp.h
index bd20531..dff1281 100644
--- a/include/linux/atomisp.h
+++ b/include/linux/atomisp.h
@@ -694,6 +694,28 @@ struct atomisp_bnlm_config  {
 	int32_t max_min_th;
 };
 
+struct atomisp_tdf_config {
+	int32_t thres_flat_table[64];
+	int32_t thres_detail_table[64];
+	int32_t epsilon_0;
+	int32_t epsilon_1;
+	int32_t eps_scale_text;
+	int32_t eps_scale_edge;
+	int32_t sepa_flat;
+	int32_t sepa_edge;
+	int32_t blend_flat;
+	int32_t blend_text;
+	int32_t blend_edge;
+	int32_t shading_gain;
+	int32_t shading_base_gain;
+	int32_t local_y_gain;
+	int32_t local_y_base_gain;
+	int32_t rad_x_origin;
+	int32_t rad_y_origin;
+	int32_t blend_ratio;
+	int32_t min_edge_thres;
+};
+
 struct atomisp_parameters {
 	struct atomisp_wb_config   *wb_config;  /* White Balance config */
 	struct atomisp_cc_config   *cc_config;  /* Color Correction config */
@@ -746,6 +768,7 @@ struct atomisp_parameters {
 	struct atomisp_macc1_5_table *macc1_5_table;
 	struct atomisp_xnr3_0_11_config *xnr3_0_11_config;
 	struct atomisp_bnlm_config  *bnlm_config;
+	struct atomisp_tdf_config *tdf_config;
 	void	*lin_2500_config;       /* Skylake: Linearization config */
 	void	*obgrid_2500_config;    /* Skylake: OBGRID config */
 	void	*bnr_2500_config;       /* Skylake: bayer denoise config */
-- 
1.9.1

